test/test_adjoint.py

```diff
@@ -10,10 +10,15 @@
 from funsor.adjoint import adjoint
 from funsor.domains import bint
 from funsor.einsum import einsum, naive_einsum, naive_plated_einsum
-from funsor.interpreter import interpretation
+from funsor.interpreter import interpretation, _GENERIC_SUBS
 from funsor.terms import Variable, reflect
 from funsor.testing import make_einsum_example, make_plated_hmm_einsum
 
+
+# TODO remove this when removing eager_subs methods
+xfail_if_new_subs = pytest.mark.xfail(_GENERIC_SUBS, reason="fails w/ new subs")
```

Comment:
The changes to `funsor.adjoint` necessary to support alpha-renaming are beyond the scope of this PR, so I've marked these tests as xfailing and will refactor in a followup PR.